lightning/src/ln/channelmanager.rs

```diff
@@ -1104,6 +1104,37 @@ impl ChannelDetails {
 	}
 }
 
+/// Used by [`ChannelManager::list_pending_payments`] to express the status of payments that are
+/// currently pending resolution. These not only include payments that are currently inflight, but
+/// also those that may have yet to find a successful path or have outstanding HTLCs.
+#[derive(Debug, PartialEq)]
+pub enum RecentPaymentDetails {
+	/// When a payment is still being sent and awaiting successful delivery.
+	Pending {
+		/// Hash of the payment that is currently being sent but has yet to be fulfilled or
+		/// abandoned.
+		payment_hash: PaymentHash,
+		/// Total amount (in msat) across all paths for this payment, not just the amount currently
+		/// inflight.
+		total_msat: u64,
+	},
+	/// When a pending payment is fulfilled, we continue tracking it until all pending HTLCs have
+	/// been resolved. Upon receiving [`Event::PaymentSent`], we delay for a few minutes before the
+	/// payment is removed from tracking.
+	Fulfilled {
+		/// Hash of the payment that was claimed. Returns `None` for possibly older serializations
+		/// of [`ChannelManager`]
+		payment_hash: Option<PaymentHash>,
```

Comment:
@TheBlueMatt Thoughts on removing the `Option` here and in `PendingOutboundPayment`? Looks like we introduced it in `0.0.104` (https://github.com/lightningdevkit/rust-lightning/commit/2c4f16d5e39e6d4e90d1b9836cceaad11810fbaa).